src/main/java/org/powerflows/dmn/io/yaml/model/rule/YamlRule.java

```diff
@@ -18,11 +18,11 @@
 
 import lombok.Data;
 
-import java.util.Map;
+import java.util.LinkedHashMap;
 
 @Data
 public final class YamlRule {
     private String description;
-    private Map<String, YamlRuleInput> in;
-    private Map<String, YamlRuleOutput> out;
+    private LinkedHashMap<String, YamlRuleEntry> in;
```

Comment:
why implementation here?

Reply:
to make snakeyaml instantiate correct type, we need to have insertion order preserved in maps

Reply:
got it, as I supposed.